File: snippets/python_snippets/send_spontaneous_payment.py
import breez_sdk
import logging

sdk_services = breez_sdk.BlockingBreezServices

def send_spontaneous_payment():
    # ANCHOR: send-spontaneous-payment
    node_id = "..."
    amount_msat = 300000
    try:
        req = breez_sdk.SendSpontaneousPaymentRequest(node_id,amount_msat)
        result = sdk_services.send_spontaneous_payment(req)
        return result
    # ANCHOR: send-spontaneous-payment
    except Exception as error:
        logging.error(error)
        raise

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/SendSpontaneousPayment.kt
package com.example.kotlinmpplib

import breez_sdk.*
class SendSpontaneousPayment {
    fun sendSpontaneousPayment(sdk: BlockingBreezServices) {
        // ANCHOR: send-spontaneous-payment
        val nodeId = "..."
        val amountMsat = 3_000_000.toULong()
        val optionalLabel = "<label>"
        try {
            val req = SendSpontaneousPaymentRequest(nodeId, amountMsat, label = optionalLabel)
            val response = sdk.sendSpontaneousPayment(req)
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: send-spontaneous-payment
    }

    fun sendSpontaneousPaymentWithTlvs(sdk: BlockingBreezServices) {
        // ANCHOR: send-spontaneous-payment-with-tlvs
        val nodeId = "..."
        val amountMsat = 3_000_000.toULong()
        val optionalExtraTlvs = listOf<TlvEntry>(
            TlvEntry(34_349_334.toULong(), "Hello world!".encodeToByteArray().asUByteArray().toList())
        )
        try {
            val req = SendSpontaneousPaymentRequest(nodeId, amountMsat, optionalExtraTlvs)
            val response = sdk.sendSpontaneousPayment(req)
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: send-spontaneous-payment-with-tlvs
    }
}